src/portable/st/synopsys/dcd_synopsys.c

```diff
@@ -248,6 +263,60 @@ static tusb_speed_t get_speed(USB_OTG_DeviceTypeDef* dev)
   return (enum_spd == DCD_HIGH_SPEED) ? TUSB_SPEED_HIGH : TUSB_SPEED_FULL;
 }
 
+
+# if defined(EMBEDDED_HS_PHY)
+static bool USB_HS_PHYCInit(void)
+{
+    USB_HS_PHYC_GlobalTypeDef *usb_hs_phyc = (USB_HS_PHYC_GlobalTypeDef *)USB_HS_PHYC_CONTROLLER_BASE;
+    // Enable LDO
+    usb_hs_phyc->USB_HS_PHYC_LDO |= USB_HS_PHYC_LDO_ENABLE;
+    int32_t count = 2000000;
+    while(1) {
+        if (usb_hs_phyc->USB_HS_PHYC_LDO & USB_HS_PHYC_LDO_STATUS)
+            break;
+        TU_ASSERT(count > 0);
+    }
+    uint32_t  phyc_pll = 0; 
+    switch (HSE_VALUE) {
+    case 12000000:
+        phyc_pll = USB_HS_PHYC_PLL1_PLLSEL_12MHZ;
+        break;
+    case 12500000:
+        phyc_pll = USB_HS_PHYC_PLL1_PLLSEL_12_5MHZ;
+        break;
+    case 16000000:
+        phyc_pll = USB_HS_PHYC_PLL1_PLLSEL_16MHZ;
+        break;
+    case 24000000:
+        phyc_pll = USB_HS_PHYC_PLL1_PLLSEL_24MHZ;   
+        break;
+    case 25000000:
+        phyc_pll = USB_HS_PHYC_PLL1_PLLSEL_25MHZ;   
+        break;
+    case 32000000:
+        // Value not defined in header
+        phyc_pll =   USB_HS_PHYC_PLL1_PLLSEL_Msk;                          ;   
+        break;
+    default:
+        TU_ASSERT(0);
+    }
+    usb_hs_phyc->USB_HS_PHYC_PLL = phyc_pll;
+
+    // Use magic value as in stm32f7xx_ll_usb.
+#  if !defined  (USB_HS_PHYC_TUNE_VALUE)
+#  define USB_HS_PHYC_TUNE_VALUE    0x00000F13U /*!< Value of USB HS PHY Tune */
+#  endif /* USB_HS_PHYC_TUNE_VALUE */
+    // Control the tuning interface of the High Speed PHY */
+    usb_hs_phyc->USB_HS_PHYC_TUNE |= USB_HS_PHYC_TUNE_VALUE;
+
+    // Enable PLL internal PHY
+    usb_hs_phyc->USB_HS_PHYC_PLL = phyc_pll | USB_HS_PHYC_PLL_PLLEN;
+
+    board_delay(2);
```

Comment:
board_delay is not accepted, if device absolutely needs delay to works, you got to do it with nop (taking into F_CPU in account) or any other way.

Reply:
Is there any hint/formula for such a  platform independant delay?

Reply:
The timeout is from the stm32f7xx_ll_usb.c.
  /* Enable PLL internal PHY */
  USB_HS_PHYC->USB_HS_PHYC_PLL |= USB_HS_PHYC_PLL_PLLEN;

No register bit mentioned to poll in in the reference manual. No need for timeout mentioned in data sheet.
For now, I could try to move USB_HS_PHYCInit to hw/bsp/stm32f723disco/stm32f723disco.c but that means code duplication for other F723 boards.

Reply:
> The timeout is from the stm32f7xx_ll_usb.c.
> /* Enable PLL internal PHY */
> USB_HS_PHYC->USB_HS_PHYC_PLL |= USB_HS_PHYC_PLL_PLLEN;
> 
> No register bit mentioned to poll in in the reference manual. No need for timeout mentioned in data sheet.

Hmm, weird, maybe st engineer want to be safe with that additional delay. We could just to see if dropping  could have any impact on the enumeration !!!

> For now, I could try to move USB_HS_PHYCInit to hw/bsp/stm32f723disco/stm32f723disco.c but that means code duplication for other F723 boards.

Yeah, that makes sense as well if you don't want to deal with it for now. Moving it to the BSP ( with few words of comment) would help to get this PR merged. Afterwards we can look more closely at the issue, I will try to open the datasheet and do an bit of testing later on.

PS: could you able to enumerate at HS with F723 and/or F769 so far ? F769 is similar to H743 (need external HS PHY), which I get running fine here, so at lest it should work for you (hopefully).